static/app/components/events/featureFlags/onboardingIntegrationSection.tsx

```diff
@@ -94,17 +105,20 @@ export default function OnboardingIntegrationSection({
             <Button
               priority="default"
               onClick={() => {
-                postSecret(secret);
-                setTokenSaved(true);
+                postSecret(secret, setSecretSaved);
               }}
               disabled={secret === ''}
```

Comment:
There's a race condition where if someone double-clicks the button we'll send out two requests which could return in either order.

For example, if the 1st request was slow for some reason, network lag, it's response would override the 2nd request. So assume the 2nd request worked properly, saved into the db. The first slow request could timeout and set `false` into state after a while. The ui would show the wrong state to the user.

Reply:
Solutions!
`useMutation` should help, and it has a `isLoading` state to disable the button while the request is in-flight.

Reply:
updated!